src/Helpers/Cache/FileSystemCacheHandler.php

```diff
@@ -19,8 +19,8 @@ class FileSystemCacheHandler implements CacheHandler
     public function __construct($temp_directory_prefix = 'auth0-php')
     {
         $this->tmp_dir = sys_get_temp_dir().DIRECTORY_SEPARATOR.$temp_directory_prefix.DIRECTORY_SEPARATOR;
-        if (! file_exists($this->tmp_dir)) {
-            mkdir($this->tmp_dir);
+        if (! is_dir($this->tmp_dir) && ! @mkdir($this->tmp_dir, 0777, true) && ! is_dir($this->tmp_dir) ) {
```

Comment:
So ... this is a lot of logic to pack into a single line here, I'd suggest breaking this out a bit to make it a little more clear what's going on here.